### PR TITLE
Daos 5625 6

### PR DESCRIPTION
**Pull request by wangdi1**

### CODE REVIEW

File: src/rebuild/srv.c [hunk 1]
```diff
@@ -823,10 +823,17 @@ rebuild_scan_broadcast(struct ds_pool *pool,
...
+
+retry:
 	rc = dss_rpc_send(rpc);
 	rso = crt_reply_get(rpc);
 	if (rc == 0)
 		rc = rso->rso_status;
+	if (rc == -DER_BUSY) {
+		D_DEBUG(DB_REBUILD, "retry broadcast scan "DF_UUID"\n",
+			DP_UUID(pool->sp_uuid));
+		goto retry;
```
Comment (liw) [inline src/rebuild/srv.c:835]: Should we dss_sleep for a short while, or that wouldn't make any sense?

**Review by liw — DISMISSED** [review]

File: src/rebuild/srv.c [hunk 2]
```diff
@@ -823,10 +823,17 @@ rebuild_scan_broadcast(struct ds_pool *pool,
...
 	crt_group_rank(pool->sp_group,  &rsi->rsi_master_rank);
+
+retry:
 	rc = dss_rpc_send(rpc);
 	rso = crt_reply_get(rpc);
 	if (rc == 0)
 		rc = rso->rso_status;
+	if (rc == -DER_BUSY) {
+		D_DEBUG(DB_REBUILD, "retry broadcast scan "DF_UUID"\n",
+			DP_UUID(pool->sp_uuid));
```
Comment (liuxuezhao) [inline src/rebuild/srv.c:834]: probably can define a max retry count to avoid possibly infinite retry

**Review by liuxuezhao — DISMISSED** [review]

File: src/rebuild/srv.c (hunk 2 above)
Comment (wangdi1) [reply src/rebuild/srv.c:834]: Thought about it, but not sure what is the best number here? and also it should expect the previous rebuild will cleanup soon anyway.

File: src/rebuild/srv.c (hunk 1 above)
Comment (wangdi1) [reply src/rebuild/srv.c:835]: oh, this will not hog cpu after this rebuild scan rpc is handled, scheduler will reschedule other ult anyway.

**Comment by wangdi1** [comment]
Just rebase with master

**Review by liw — APPROVED** [review]

File: src/rebuild/srv.c (hunk 1 above)
Comment (liw) [reply src/rebuild/srv.c:835]: Compared to hogging the local CPU, I was more concerned about sending at full speed the retry RPCs, most of which end up getting -DER_BUSY.

Comment (wangdi1) [reply src/rebuild/srv.c:835]: hmm, this stage is supposed to be very short.  but probably adding a ult sleep would not hurt here. I will retry this later. thanks.

**Review by liuxuezhao — APPROVED** [review]

**Pull request merged**